test/callgrind_test.cpp

```diff
@@ -29,89 +26,63 @@ class callgrind_test : public beast::unit_test::suite
-            saltValue,
-            sizeof(key_type),
-            nudb::block_size(path),
-            0.50,
-            ec);
-        if(! expect(! ec, ec.message()))
+        test_store ts{sizeof(key_type), blockSize, loadFactor};
+        ts.create(ec);
+        if(! BEAST_EXPECTS(! ec, ec.message()))
             return;
-        finisher f(
-            [&]
-            {
-                {
-                    error_code ev;
-                    native_file::erase(dp, ev);
-                    expect(! ev, ev.message());
-                }
-                {
-                    error_code ev;
-                    native_file::erase(kp, ev);
-                    expect(! ev, ev.message());
-                }
-                {
-                    error_code ev;
-                    erase_file(lp, ev);
-                    expect(! ev, ev.message());
-                }
-            });
-        store db;
-        db.open(dp, kp, lp, arenaAllocSize, ec);
-        if(! expect(! ec, ec.message()))
+        ts.open(ec);
+        if(! BEAST_EXPECTS(! ec, ec.message()))
             return;
-        expect(db.appnum() == appnumValue, "appnum");
-        Sequence seq;
-        for(std::size_t i = 0; i < count; ++i)
+        for(std::size_t i = 0; i < N; ++i)
         {
-            auto const v = seq[i];
-            auto const success = db.insert(
-                &v.key, v.data, v.size, ec);
-            if(! expect(! ec, ec.message()))
+            auto const item = ts[i];
+            ts.db.insert(item.key, item.data, item.size, ec);
+            if(! BEAST_EXPECTS(! ec, ec.message()))
                 return;
-            expect(success);
         }
-        Storage s;
-        for(std::size_t i = 0; i < count * 2; ++i)
+        Buffer b;
+        for(std::size_t i = 0; i < N * 2; ++i)
         {
             if(! (i%2))
             {
-                auto const v = seq[i/2];
-                expect (db.fetch(&v.key, s, ec), "fetch");
-                if(! expect(! ec, ec.message()))
+                auto const item = ts[i/2];
+                ts.db.fetch(item.key, b, ec);
+                if(! BEAST_EXPECTS(! ec, ec.message()))
+                    return;
+                if(! BEAST_EXPECT(b.size() == item.size))
+                    return;
+                if(! BEAST_EXPECT(std::memcmp(b.data(),
+                        item.data, item.size) == 0))
                     return;
-                expect (s.size() == v.size, "size");
-                expect (std::memcmp(s.get(),
-                    v.data, v.size) == 0, "data");
             }
             else
             {
-                auto const v = seq[count + i/2];
-                auto const found = db.fetch (&v.key, s, ec);
-                if(! expect(! ec, ec.message()))
+                auto const item = ts[N + i/2];
+                ts.db.fetch(item.key, b, ec);
+                if(! BEAST_EXPECTS(ec ==
+                        error::key_not_found, ec.message()))
```

Comment:
You mean the test will fail if **any error but** `key_not_found` occurs, including no error. And I believe this behavior is correct but I'm open to hearing opposing arguments.